internal-webapps/org.wso2.bfsi.authentication.endpoint/src/main/java/org/wso2/bfsi/authentication/endpoint/BFSIConsentServlet.java

```diff
@@ -0,0 +1,259 @@
+     */
+    JSONObject createConsentDataset(JSONObject consentResponse, int statusCode) throws IOException {
+
+        JSONObject errorObject = new JSONObject();
+        if (statusCode != HttpURLConnection.HTTP_OK) {
+            if (statusCode == HttpURLConnection.HTTP_UNAUTHORIZED) {
+                if (consentResponse.has(Constants.DESCRIPTION)) {
+                    errorObject.put(Constants.IS_ERROR, consentResponse.get(Constants.DESCRIPTION));
+                }
+            } else {
+                errorObject.put(Constants.IS_ERROR, "Retrieving consent data failed");
+            }
+            return errorObject;
+        } else {
+            return consentResponse;
+        }
+    }
+
+    /**
+     * Retrieve the config.
+     */
+    void setAuthExtension() {
+
+        try {
+            bfsiAuthServletTK = (BFSIAuthServletInterface) Class.forName(ConsentManagementConfigParser.getInstance().
+                    getAuthServletExtension()).getDeclaredConstructor().newInstance();
+        } catch (InstantiationException | IllegalAccessException |
+                InvocationTargetException | NoSuchMethodException | ClassNotFoundException e) {
+            log.error("Webapp extension not found", e);
+        }
+    }
+
+    @Generated(message = "Encapsulated method for unit test")
+    ResourceBundle getResourceBundle(Locale locale) {
+
+        return ResourceBundle.getBundle(BUNDLE, locale);
+    }
+
+    /**
+     * Retrieve admin credentials in Base64 format from webapp properties or OB configs.
+     */
+    static String getConsentApiCredentials () {
+        String username, password;
+        try {
+            InputStream configurations = BFSIConsentConfirmServlet.class.getClassLoader()
+                    .getResourceAsStream(Constants.CONFIG_FILE_NAME);
+            Properties configurationProperties = new Properties();
+            configurationProperties.load(configurations);
+            boolean isConfiguredInWebapp = Boolean.parseBoolean(
+                    configurationProperties.getProperty(Constants.LOCATION_OF_CREDENTIALS));
+            if (!isConfiguredInWebapp) {
+                username = ConsentManagementConfigParser.getInstance().getConsentAPIUsername();
+                password = ConsentManagementConfigParser.getInstance().getConsentAPIPassword();
+            } else {
+                username = configurationProperties.getProperty(Constants.USERNAME_IN_WEBAPP_CONFIGS);
+                password = configurationProperties.getProperty(Constants.PASSWORD_IN_WEBAPP_CONFIGS);
+            }
+        } catch (IOException | NullPointerException e) {
+            log.error("Error occurred while reading the webapp properties file. Therefore using OB configurations.");
+            username = ConsentManagementConfigParser.getInstance().getConsentAPIUsername();
+            password = ConsentManagementConfigParser.getInstance().getConsentAPIPassword();
```

Comment:
Shall we keep the config parser instance in a class variable?

Reply:
Fixed in https://github.com/wso2-extensions/bfsi-consent-management/pull/8/commits/7c13892e0735c7d32a51565d4369ab1e9eddbc7e